### Question 1
  
What's the version of pipenv you installed?  
2024.2.0

### Question 2
  
What's the first hash for scikit-learn you get in Pipfile.lock?

```console
pipenv install scikit-learn==1.5.2
```

"sha256:03b6158efa3faaf1feea3faa884c840ebd61b6484167c711548fce208ea09445"

### Question 3

In [1]:
import pickle

In [ ]:
with open('model1.bin', 'rb') as model_file:
    model = pickle.load(model_file)

with open('dv.bin', 'rb') as dv_file:
    dv = pickle.load(dv_file)

In [1]:
client = {"job": "management", "duration": 400, "poutcome": "success"}

In [12]:
X = dv.transform(client)

round(float(model.predict_proba(X)[0, 1]), 3)

0.759

What's the probability that this client will get a subscription?
0.759

### Question 4

In [2]:
# put this in a file can run it
import pickle
from flask import Flask, request, jsonify

with open('model1.bin', 'rb') as model_file:
    model = pickle.load(model_file)

with open('dv.bin', 'rb') as dv_file:
    dv = pickle.load(dv_file)

app = Flask('Predict')

@app.route('/predict', methods=['POST'])
def predict():
    client = request.get_json()

    X = dv.transform(client)
    y_probability = model.predict_proba(X)[0, 1]

    y_prediction = y_probability >= 0.5

    result = {
        "probability": float(y_probability),
        "prediction": bool(y_prediction)
    }

    return jsonify(result)

if __name__ == "__main__":
    app.run(debug=True, host='localhost', port=9696)

In [ ]:
import requests

url = 'http://127.0.0.1:9696/predict'
client = {"job": "student", "duration": 280, "poutcome": "failure"}

requests.post(url, json=client).json()

```
gunicorn --bind 127.0.0.1:9696 predict:app
```

What's the probability that this client will get a subscription?  
  
**0.335**  
0.535  
0.735  
0.935

### Question 5

```
docker pull svizor/zoomcamp-model:3.11.5-slim

docker images
```

So what's the size of this base image?  
  
45 MB  
**130 MB**  
245 MB  
330 MB